src/server/game/Spells/Spell.cpp

```diff
@@ -3449,6 +3460,12 @@ void Spell::_cast(bool skipCheck)
 template <class Container>
 void Spell::DoProcessTargetContainer(Container& targetContainer)
 {
+    for (TargetInfoBase& target : targetContainer)
+    {
+        if (target.MissedTarget(this))
+            return;
```

Comment:
This will cause any target scheduled processing after the first one that fails to fail.